src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -0,0 +1,116 @@
+import {Directive, ElementRef, Input, ViewContainerRef, OnDestroy} from '@angular/core';
+import {Overlay, OverlayRef, OverlayState, TemplatePortal} from '../core';
+import {MdAutocomplete} from './autocomplete';
+import {PositionStrategy} from '../core/overlay/position/position-strategy';
+import {Observable} from 'rxjs/Observable';
+import {Subscription} from 'rxjs/Subscription';
+import 'rxjs/add/observable/merge';
+
+/** The panel needs a slight y-offset to ensure the input underline displays. */
+export const MD_AUTOCOMPLETE_PANEL_OFFSET = 6;
+
+@Directive({
+  selector: 'input[mdAutocomplete], input[matAutocomplete]',
+  host: {
+    '(focus)': 'openPanel()'
+  }
+})
+export class MdAutocompleteTrigger implements OnDestroy {
+  private _overlayRef: OverlayRef;
+  private _portal: TemplatePortal;
+  private _panelOpen: boolean = false;
+  private _closeWatcher: Subscription;
+
+  /* The autocomplete panel to be attached to this trigger. */
+  @Input('mdAutocomplete') autocomplete: MdAutocomplete;
+
+  constructor(private _element: ElementRef, private _overlay: Overlay,
+              private _vcr: ViewContainerRef) {}
+
+  ngOnDestroy() { this.destroyPanel(); }
+
+  /* Whether or not the autocomplete panel is open. */
+  get panelOpen(): boolean {
+    return this._panelOpen;
+  }
+
+  /** Opens the autocomplete suggestion panel. */
+  openPanel(): void {
+    if (!this._overlayRef) {
+      this._createOverlay();
+    }
+
+    if (!this._overlayRef.hasAttached()) {
+      this._overlayRef.attach(this._portal);
+      this._watchForClose();
+    }
+
+    this._panelOpen = true;
+  }
+
+  /** Closes the autocomplete suggestion panel. */
+  closePanel(): void {
+    if (this._overlayRef && this._overlayRef.hasAttached()) {
+      this._overlayRef.detach();
+    }
+
+    this._closeWatcher.unsubscribe();
+    this._panelOpen = false;
+  }
+
+  /** Destroys the autocomplete suggestion panel. */
+  destroyPanel(): void {
```

Comment:
Hmm, someone could possibly call it if they knew that the autocomplete wouldn't be used anymore for some reason and they had a ton of options. But I don't really think it's that useful. Removing.